### PR TITLE
Panther docker

### PR DESCRIPTION
**Pull request by KmakD**
Initial draft for repo

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.vscode
+__pycache__/
```
Comment (Kotochleb) [inline .gitignore:2]: Can be replaced with [.gitignore](https://github.com/atlassian/docker/blob/master/.gitignore)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+RUN mkdir build && \
+    source /opt/ros/$ROS_DISTRO/setup.bash && \
+    rosdep init && \
+    rosdep update --rosdistro $ROS_DISTRO && \
+    catkin_make -DCATKIN_ENABLE_TESTING=0 -DCMAKE_BUILD_TYPE=Release
+
+# Clear 
+RUN apt clean \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (Kotochleb) [inline Dockerfile:50]: This step has to be right after third step

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+RUN git clone https://github.com/byq77/apa102-pi.git src/apa102-pi  && \
+    cd src/apa102-pi && sudo python3 setup.py install
+
+RUN git clone --branch panther_ros https://github.com/husarion/panther_ros.git src/panther_ros  && \
+    vcs import src < src/panther_ros/panther/panther.repos
+
+RUN mkdir build && \
+    source /opt/ros/$ROS_DISTRO/setup.bash && \
+    rosdep init && \
+    rosdep update --rosdistro $ROS_DISTRO && \
```
Comment (Kotochleb) [inline Dockerfile:45]: rosdep automatically runs `apt update` so
``` bash
apt clean
rm -rf /var/lib/apt/lists/* 
```
is needed afterwards.

File: compose.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+services:
+  ros_master:
+    image: ros:noetic-ros-core
+    tty: true
+    container_name: ros_master
+    network_mode: host
+    ipc: host
```
Comment (Kotochleb) [inline compose.yaml:7]: This line is not needed

File: compose.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    environment:
+      - ROS_IP=10.15.20.2
+    command: roscore
+
+  panther_ros:
+    build:
+      context: .
+    container_name: panther_ros
+    network_mode: host
+    ipc: host
```
Comment (Kotochleb) [inline compose.yaml:18]: This line is not needed

File: compose.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    restart: always
+    environment:
+      - ROS_IP=10.15.20.2
+    command: roscore
+
+  panther_ros:
+    build:
+      context: .
```
Comment (Kotochleb) [inline compose.yaml:15]: It might use prebuilt docker

File: compose.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      - ~/.ssh/id_rsa:/root/.ssh/id_rsa
+      - /tmp/panther_config.yaml:/tmp/panther_config.yaml
+    command: roslaunch panther_bringup bringup.launch --wait
+
+  webui-ros-joystick:
+    image: husarion/webui-ros-joystick:noetic
+    tty: true
+    container_name: webui-ros-joystick
+    network_mode: host
+    ipc: host
```
Comment (Kotochleb) [inline compose.yaml:44]: This line is not needed

**Review by Kotochleb — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (KmakD) [reply Dockerfile:50]: :+1:

File: Dockerfile (hunk 3 above)
Comment (KmakD) [reply Dockerfile:45]: :+1:

File: compose.yaml (hunk 4 above)
Comment (KmakD) [reply compose.yaml:7]: :+1:

File: .gitignore (hunk 1 above)
Comment (KmakD) [reply .gitignore:2]: :+1:

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    catkin_make -DCATKIN_ENABLE_TESTING=0 -DCMAKE_BUILD_TYPE=Release && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY ./ros_entrypoint.sh / 
+ENTRYPOINT ["/ros_entrypoint.sh"]
+
+CMD ["bash"]
```
Comment (Kotochleb) [inline Dockerfile:58]: This line can be removed

**Review by Kotochleb — CHANGES_REQUESTED** [review]

**Review by Kotochleb — APPROVED** [review]

**Pull request merged**